server/events/command_runner_test.go

```diff
@@ -234,3 +238,20 @@ func TestRunAutoplanCommand_DeletePlans(t *testing.T) {
 	ch.RunAutoplanCommand(fixtures.GithubRepo, fixtures.GithubRepo, fixtures.Pull, fixtures.User)
 	pendingPlanFinder.VerifyWasCalledOnce().DeletePlans(tmp)
 }
+
+func TestApplyWithAutoMerge_VSCMerge(t *testing.T) {
+	t.Log("if \"atlantis apply\" is run with automerge and at least one project" +
+		" has a discarded plan, automerge should not take place")
+
+	vcsClient := setup(t)
+	pull := &github.PullRequest{
+		State: github.String("open"),
+	}
+	modelPull := models.PullRequest{State: models.OpenPullState}
+	When(githubGetter.GetPullRequest(fixtures.GithubRepo, fixtures.Pull.Num)).ThenReturn(pull, nil)
+	When(eventParsing.ParseGithubPull(pull)).ThenReturn(modelPull, modelPull.BaseRepo, fixtures.GithubRepo, nil)
+	ch.GlobalAutomerge = true
+
+	ch.RunCommentCommand(fixtures.GithubRepo, &fixtures.GithubRepo, nil, fixtures.User, fixtures.Pull.Num, &events.CommentCommand{Name: models.ApplyCommand})
+	vcsClient.VerifyWasCalledOnce().MergePull(modelPull)
```

Comment:
This test should test that `MergePull` is **not** called.

Reply:
There was no test at all on MergePull to begin with afaics
This test, tests that the `mergepull` feature actually works.
I wanted to provide the test that shows `mergepull` is not called when a plan is discarded, but didn't manage to get this to work with stubbing. See my prev comment.

To be clear, do you suggest I use an actual BoltDB from a file I need to create, that will have one project in the `discardedPlan` state? So I can assert that `mergepull` is not called?

Reply:
Looks like the only test was `TestRunAutoplanCommand_DeletePlans` so it makes sense to add this automerge test. Good catch. For testing the new functionality, could you seed the DB with your new `UpdateProjectStatus` function? I can't see your comment from a few days ago unfortunately, looks like it's lost. What did it say?

Reply:
Oh nvm, I see it now.

Reply:
I think something like this should work:
```go
func TestRunApply_DiscardedProjects(t *testing.T) {
	vcsClient := setup(t)
	ch.GlobalAutomerge = true
	defer func() { ch.GlobalAutomerge = false }()
	tmp, cleanup := TempDir(t)
	defer cleanup()
	boltDB, err := db.New(tmp)
	Ok(t, err)
	ch.DB = boltDB
	pull := fixtures.Pull
	pull.BaseRepo = fixtures.GithubRepo
	_, err = boltDB.UpdatePullWithResults(pull, []models.ProjectResult{
		{
			Command:    models.PlanCommand,
			RepoRelDir: ".",
			Workspace:  "default",
			PlanSuccess: &models.PlanSuccess{
				TerraformOutput: "tf-output",
				LockURL:         "lock-url",
			},
		},
	})
	Ok(t, err)
	Ok(t, boltDB.UpdateProjectStatus(pull, "default", ".", models.DiscardedPlanStatus))
	ghPull := &github.PullRequest{
		State: github.String("open"),
	}
	When(githubGetter.GetPullRequest(fixtures.GithubRepo, fixtures.Pull.Num)).ThenReturn(ghPull, nil)
	When(eventParsing.ParseGithubPull(ghPull)).ThenReturn(pull, pull.BaseRepo, fixtures.GithubRepo, nil)
	When(workingDir.GetPullDir(matchers.AnyModelsRepo(), matchers.AnyModelsPullRequest())).
		ThenReturn(tmp, nil)
	ch.RunCommentCommand(fixtures.GithubRepo, &fixtures.GithubRepo, &pull, fixtures.User, fixtures.Pull.Num, &events.CommentCommand{Name: models.ApplyCommand})
	vcsClient.VerifyWasCalled(Never()).MergePull(matchers.AnyModelsPullRequest())
}
```

Reply:
Awesome looking good, thanks 👍